lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -192,6 +192,27 @@ def retrying_submit(actions)
       end
     end
 
+    # we shouldn't send unsupported actions to ES
+    # method filters out unsupported actions by warning, writes event to DQL if enabled
+    # @returns valid actions
+    def filter_unsupported_actions(actions)
+      return if actions.nil? || actions.size < 1
+      supported_actions, unsupported_actions = actions.partition { |action, _, _| LogStash::Outputs::ElasticSearch::VALID_HTTP_ACTIONS.include?(action) }
+      if !unsupported_actions.nil? && unsupported_actions.size > 0
+        handle_dlq_or_drop(unsupported_actions)
+        @logger.warn("Number of requests filtered out before sending to Elasticsearch because of unsupported action, ", size: actions.size)
+      end
+      supported_actions
+    end
+
+    def handle_dlq_or_drop(unsupported_actions)
```

Comment:
nitpick: method name doesn't indicate that this has semantics specific to unsupported actions, so we risk accidental use in later development